### ISSUE TITLE
pyjnius build error on s390x

### ISSUE THREAD

**Issue by ravigumm** [kivy/pyjnius#663]
In pyjnius 1.5.0, with cython 0.29.35, I am getting the following error when trying to install pyjnius from source code on s390x `unknown file type '.pxi'`
```
pyjnius]# pip install .
Processing /root/ravi/pyjnius
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: pyjnius
  Building wheel for pyjnius (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pyjnius (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      warning: [options] bootstrap class path not set in conjunction with -source 1.7
      1 warning
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-s390x-cpython-310
      copying jnius_config.py -> build/lib.linux-s390x-cpython-310
      copying setup_sdist.py -> build/lib.linux-s390x-cpython-310
      creating build/lib.linux-s390x-cpython-310/jnius
      copying jnius/__init__.py -> build/lib.linux-s390x-cpython-310/jnius
      copying jnius/env.py -> build/lib.linux-s390x-cpython-310/jnius
      copying jnius/reflect.py -> build/lib.linux-s390x-cpython-310/jnius
      copying jnius/signatures.py -> build/lib.linux-s390x-cpython-310/jnius
      creating build/lib.linux-s390x-cpython-310/jnius/src
      creating build/lib.linux-s390x-cpython-310/jnius/src/org
      creating build/lib.linux-s390x-cpython-310/jnius/src/org/jnius
      copying jnius/src/org/jnius/NativeInvocationHandler.java -> build/lib.linux-s390x-cpython-310/jnius/src/org/jnius
      copying jnius/src/org/jnius/NativeInvocationHandler.class -> build/lib.linux-s390x-cpython-310/jnius/src/org/jnius
      running build_ext
      warning: jnius/config.pxi:1:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/jnius.pyx:102:0: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/config.pxi:1:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      Compiling jnius/jnius.pyx because it changed.
      [1/1] Cythonizing jnius/jnius.pyx
      building 'jnius' extension
      error: unknown file type '.pxi' (from 'jnius/jni.pxi')
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyjnius
```

**Comment by cmacdonald** [comment]
Can you try commenting out all the .pxi files in FILES in setup.py, i.e.:

```python
FILES = [
    # 'jni.pxi',
    # 'jnius_compat.pxi',
    # 'jnius_conversion.pxi',
    # 'jnius_export_class.pxi',
    # 'jnius_export_func.pxi',
    # 'jnius_jvm_android.pxi',
    # 'jnius_jvm_desktop.pxi',
    # 'jnius_jvm_dlopen.pxi',
    # 'jnius_localref.pxi',
    # 'jnius_nativetypes3.pxi',
    # 'jnius_proxy.pxi',
    'jnius.pyx',
    # 'jnius_utils.pxi',
]
```

**Comment by Cheaterman** [comment]
The above should work, alternatively you can use Cython 0.29.36.

**Comment by ravigumm** [comment]
Commenting out pxi files in setup.py works for me. Thanks

**Comment by ravigumm** [comment]
Just updating cython from 0.29.35 to 0.29.36 did not help surpass the error about pxi files. I had to comment out those lines in setup.py to surpass the error. Thanks

**Comment by pratapkishorevarma** [comment]
> Can you try commenting out all the .pxi files in FILES in setup.py, i.e.:
> 
> ```python
> FILES = [
>     # 'jni.pxi',
>     # 'jnius_compat.pxi',
>     # 'jnius_conversion.pxi',
>     # 'jnius_export_class.pxi',
>     # 'jnius_export_func.pxi',
>     # 'jnius_jvm_android.pxi',
>     # 'jnius_jvm_desktop.pxi',
>     # 'jnius_jvm_dlopen.pxi',
>     # 'jnius_localref.pxi',
>     # 'jnius_nativetypes3.pxi',
>     # 'jnius_proxy.pxi',
>     'jnius.pyx',
>     # 'jnius_utils.pxi',
> ]
> ```

What is the long term solution for this issue instead of manually commenting the pxi files?
Is there a plan to fix this?

**Comment by cmacdonald** [comment]
> I believe the plan might be to remove those lines from pyjnius itself

We should do this in a new release asap, as it may prevent jnius installation on platforms without prebuilt binaries.

I have worked building my own jnius in the past, but hadn't seen this problem before. I guess something changed in cython in the last year or two, but not that recently.

**Comment by cmacdonald** [comment]
I think this is now addressed by #669. 
Would you be able to try to install the current Git please @ravigumm?

**Comment by misl6** [comment]
Fixed via #669 .

Please upgrade to `1.6.0` and open a new issue if the issue persists on your side.

**Comment by BH801** [comment]
Cython==0.29.23，pyjnius==1.4.2    isok
